docs/roadmap/rm-2020.md

```diff
@@ -0,0 +1,79 @@
+# 2020 Eventing Roadmap
+
+## References
+
+[Personas and User Stories](../personas.md):
+
+Knative Eventing caters to the following personas when developing different
+features
+
+1. Event consumer (developer)
+1. Event producer
+1. System Integrator
+1. Contributors
+
+## Areas of Interest and Requirements
+
+1. **API Maturity**. Description TBD
+1. **Production-ready Brokers**. Description TBD
+1. **Knative Inbound Authorization**. Description TBD
+1. **Serverless Ecosystem**. Description TBD
+1. **Test and Release**. Description TBD
+1. **Multi-Tenancy**. Description TBD
+
+### API Maturity
+
+1. **Define Broker spec and conformance tests** including data plane
+   specification https://github.com/knative/eventing/issues/2614. (issue TBD)
+1. **Graduate to V1 APIs** This also mplies increasing of conformance test for
+   channels, broker and sources. (issue TBD)
+
+### Production-ready Brokers
+
+1. **Minimal Footprint (CPU/RAM)** The underlying platform components have a
+   minimal (near zero) footprint (CPU/RAM) in the cluster when eventing is not
+   being used. (issue TBD)
+1. **Observability** Improvements to E2E tracing, metrics, and logs. (issue TBD)
+1. **Lossless upgrades** Brokers never go unavailable during upgrade and no
+   events get lost. (issue TBD)
+
+### Knative Inbound Authorization
+
+1. **Application Level Authorization** (Not limited to eventing) Standardize a
+   Knative way to do application level authorization and make the policy
+   implementation/provider extensible (e.g. GCP IAM, Azure AD, etc.). (issue
+   TBD)
+1. **Unauthorized Requests Denial** (Not limited to eventing) For Knative
+   services, unauthorized requests will not trigger user pods creation.(issue
+   TBD)
+1. **Events Authorization Policies** Any “authorizable” service will be able to
+   create policies to control what events can be accepted.(issue TBD)
+1. **Securing Eventing Documentation** Documentation of best practices for
+   securing an eventing environment (likely combined with service mesh and
+   network policy).(issue TBD)
+
+### Serverless Ecosystem
+
+1. **More Sources** Make it easier to write sources. (issue TBD)
+1. **Data Plane Scalability** TBD. (issue TBD)
+1. **Second party events** Integration with registry and discovery
+   (CloudEvents). (issue TBD)
+1. **Go, and Java CloudEvents sdks** Golang and Java CloudEvents SDKs should be
+   in a good shape both to implement eventing components and to be used by the
+   end user to interact with eventing. (issue TBD)
+
+### Test and Release
+
+1. **Robust CI/CD release pipeline** Upgrade tests for Knative Eventing
+   (Stability). (issue TBD)
+
```

Comment:
I would add a couple of points here which IMO are crucial (not sure how they should be organized)

1. Simplify environment bootstrap of e2e tests: contributors should be able to prepare an e2e tests bootstrap with one command the e2e tests, eventually starting a local cluster if needed (using minikube, or kind, or whatever). Actual scripts are mostly designed to run on CI and do assumptions on the environment, they can't be always _sourceable_ and they don't allow to select the components to install for testing 
1. Simplify/document interactive debugging for e2e tests: https://github.com/knative/eventing/issues/1733
1. Setup CI to run microbenchmarks (this is already progressing with https://github.com/knative/test-infra/issues/1907 and https://github.com/knative/test-infra/issues/1906)
1. Increase microbenchmarks coverage of data-plane hot paths